taskvine/src/worker/vine_worker.c

```diff
@@ -208,6 +211,66 @@ __attribute__((format(printf, 2, 3))) void send_message(struct link *l, const ch
 	va_end(va);
 }
 
+/* Send messages from list of asychronous messages available*/
+
+void deliver_async_messages(struct link *l)
+{
+	int bytes_in_buffer = get_link_buffer_bytes(l);
+	int bytes_available = VINE_MAX_BUFFER_SIZE - bytes_in_buffer;
```

Comment:
IMHO, `VINE_MAX_BUFFER_SIZE` is not a protocol constant, but the known size of the TCP window size.  This is 64KB by default but can be set/get with `link_window_set/get`.